### PR TITLE
Make SonataEasyExtends optional

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is BC.

Part of: https://github.com/sonata-project/dev-kit/issues/750

SonataEasyExtends is deprecated but it is used on many bundles, this makes it optional. This change is BC because if the user does not change anything it will continue using the deprecated code but with a message.

## Changelog

```markdown
### Changed
- SonataEasyExtendsBundle is now optional, using SonataDoctrineBundle is preferred
### Deprecated
- Using SonataEasyExtendsBundle to add Doctrine mapping information
```

## To do

- [x] Update the documentation;
- [x] Add an upgrade note.

### CODE REVIEW

File: docs/reference/installation.rst [hunk 1]
```diff
@@ -42,36 +41,11 @@ are not already enabled::
...
-    public function registerBundles()
-    {
-        return [
-            new Sonata\AdminBundle\SonataAdminBundle(),
-            new Sonata\CoreBundle\SonataCoreBundle(),
-            new Sonata\BlockBundle\SonataBlockBundle(),
-            new Sonata\EasyExtendsBundle\SonataEasyExtendsBundle(),
-            // ...
-            new FOS\UserBundle\FOSUserBundle(),
-            new Sonata\UserBundle\SonataUserBundle(),
```
Comment (core23) [inline docs/reference/installation.rst:70]: Why did you remove this?

File: docs/reference/installation.rst [hunk 2]
```diff
@@ -42,36 +41,11 @@ are not already enabled::
...
-
-    // app/AppKernel.php
-
-    public function registerBundles()
-    {
-        return [
-            new Sonata\AdminBundle\SonataAdminBundle(),
-            new Sonata\CoreBundle\SonataCoreBundle(),
-            new Sonata\BlockBundle\SonataBlockBundle(),
-            new Sonata\EasyExtendsBundle\SonataEasyExtendsBundle(),
```
Comment (core23) [inline docs/reference/installation.rst:67]: Please add a note for `SonataDoctrineBundle`

File: docs/reference/installation.rst (hunk 1 above)
Comment (jordisala1991) [reply docs/reference/installation.rst:70]: It is still above on the docs, the recommend way on sf 4.4 is the config/bundles.php

File: docs/reference/installation.rst [hunk 3]
```diff
@@ -1,21 +1,32 @@
...
 
 Installation
 ============
 
 Prerequisites
 -------------
 
-PHP 7 and Symfony >=3.4 or >= 4.2 are needed to make this bundle work, there are
-also some Sonata dependencies that need to be installed and configured beforehand:
+PHP 7.2 and Symfony 4.4 are needed to make this bundle work, there are
```
Comment (phansys) [inline docs/reference/installation.rst:11]: IMO, if we don't use range operators or explain that these are the minimum requirements, the message is telling that only these specific versions are allowed.
```suggestion
PHP ^7.2 and Symfony ^4.4 are needed to make this bundle work, there are
```

Comment (core23) [reply docs/reference/installation.rst:11]: The version in the docs soon become outdated. Can't we just remove the versions? Composer will tell what's missing.

Comment (phansys) [reply docs/reference/installation.rst:11]: >The version in the docs soon become outdated. Can't we just remove the versions? Composer will tell what's missing.

I understand the point and agree, but I think this may lead to a poor experience for newcomers if they are not confident executing a command before being sure if the requirements are met.
Maybe we could aim the users to execute `composer require ...` and optionally target to https://packagist.org/packages/sonata-project/user-bundle in case they want to know about the requirements, since in this case Packagist will act as a documentation for our constraints.

File: docs/reference/installation.rst [hunk 4]
```diff
@@ -25,141 +36,212 @@ installation chapter.
...
-    composer update
+.. note::
+
+    This will install the FOSUserBundle_, too.
+
+If you want to use the REST API, you also need ``friendsofsymfony/rest-bundle`` and ``nelmio/api-doc-bundle``::
 
-Next, be sure to enable the bundles in your ``bundles.php`` file if they
+    composer require nelmio/api-doc-bundle
+    composer require friendsofsymfony/rest-bundle
```
Comment (phansys) [inline docs/reference/installation.rst:50]:
```suggestion
    composer require friendsofsymfony/rest-bundle nelmio/api-doc-bundle
```

Comment (jordisala1991) [reply docs/reference/installation.rst]: Done

File: UPGRADE-4.x.md [hunk 5]
```diff
@@ -1,12 +1,26 @@
...
+=========================
+
 ### Deprecated `TwoStepVerificationCommand` with no arguments
 
 Attempting to run that command with 2FA enabled and without building the
 command with its `$helper` and `$userManager` arguments beforehand is
 deprecated. Make sure you provide both arguments.
 
+UPGRADE FROM 4.1 to 4.1.1
+=========================
```
Comment (greg0ire) [inline UPGRADE-4.x.md:22]: I forgot to do my job… it's weird that the heading mentions a patch version though, we are not supposed to introduce deprecations in patch versions…

Comment (jordisala1991) [reply UPGRADE-4.x.md:22]: And it is another PR of mine, I think I introduced a deprecation on a patch, and without the proper changelog, because I could not find one for that release

**Comment by greg0ire** [comment]
Thanks @jordisala1991 !
